parse_emails/handle_eml.py

```diff
@@ -423,34 +423,58 @@ def mime_decode(word_mime_encoded):
     return prefix + byte_string.decode(charset) + suffix
 
 
-def get_email_address(eml, entry):
+def get_email_address(eml, header_name):
     """
-    This function gets email addresses from an eml object, i.e eml[entry].
+    This function gets email addresses from an eml object, i.e eml[header_name].
     Args:
-        eml : Email object.
-        entry (str) : entry to look for in the email. i.e ('To', 'CC', 'From')
+        eml: Email object.
+        header_name (str): header to look for in the email. i.e ('To', 'CC', 'From')
     Returns:
-        res (str) : string of all required email addresses.
+        res (str): string of all required email addresses.
     """
-    if entry == 'from':
-        gel_all_values_from_email_by_entry = [str(current_eml_no_newline).replace('\r\n', '').replace('\n', '')
-                                              for current_eml_no_newline in eml.get_all(entry, [])]
+    if header_name in ['from']:
+        header_value = [str(current_eml_no_newline).replace('\r\n', '').
+                        replace('\n', '').replace('\t', '')
+                        for current_eml_no_newline in eml.get_all(header_name, [])]
     else:
-        gel_all_values_from_email_by_entry = eml.get_all(entry, [])
-    try:
-        for index, address in enumerate(gel_all_values_from_email_by_entry):
-            if 'unknown-8bit' in address:
-                updated_address = email.header.make_header(email.header.decode_header(address))
-                gel_all_values_from_email_by_entry[index] = str(updated_address)
+        header_value = eml.get_all(header_name, [])
+
+    res = []
+    addresses = header_value[0].split(',') \
+        if header_value else header_value
+    for address in addresses:
+        if 'unknown-8bit' in address:
+            try:
+                decoded = str(make_header(decode_header(address)))
+                result = get_address(decoded)
+            except UnicodeDecodeError:
+                email_address = get_address(address)
+                result = str(make_header(decode_header(email_address)))
```

Comment:
IS there any other reason to get `UnicodeDecodeError`?
If so and the actions in the exceptions can fail as well, we might need another try-except here.
if needed something like:

```suggestion
            except UnicodeDecodeError:
                try:
                    email_address = get_address(address)
                    result = str(make_header(decode_header(email_address)))
                except:
                    logger.debug(f'Fail decoding address {address=}')
```